### ISSUE TITLE
`useSelect` merges `label` and `aria-label`

### ISSUE THREAD

**Issue by jguddas** [adobe/react-spectrum#4920]
### Provide a general summary of the issue here

When you use `useSelect({ label: 'foo', aria-label: 'bar' })` the trigger prop becomes the label `foo bar`.

### 🤔 Expected Behavior?

the screen reader should read out `bar`

### 😯 Current Behavior

the screen reader reads out `foo bar`

### 💁 Possible Solution

don't merge labels

### 🔦 Context

You might have a select that has emojis or similar in the label, in that case you would like to create a screen reader friendly alternative.

This is currently not possible.

### 🖥️ Steps to Reproduce

https://codesandbox.io/s/strange-sound-gmlr66

### Version

3.16.1

### What browsers are you seeing the problem on?

Chrome

### What operating system are you using?

Linux

**Comment by jguddas** [comment]
Looks like `useMenuTrigger` could be the culprit.

https://github.com/adobe/react-spectrum/blob/09a258f098147f7e67a0430b593b08168220ec4a/packages/%40react-aria/menu/src/useMenuTrigger.ts#L137-L138

**Comment by snowystinger** [comment]
You can also hide specific parts of the visual label in other ways, such as using aria-hidden
https://codesandbox.io/s/musing-water-mx9cwv?file=/src/App.tsx:1305-1366

or linking another label via aria-labelledby and more specifically targeting the text node and not the emoji
```
<span><span id="read-this-part">Pick your favorite animal</span><span>🐹</span></span>
<Select aria-labelledby="read-this-part" />
```

**Comment by jguddas** [comment]
I know that there are workarounds such as `<span aria-hidden="true">🐹</span>`.

It's just that the default behaviour for all other elements including native elements such as the html select is that if you put an aria label it does not get merged.

**Comment by snowystinger** [comment]
I think we could try changing this line
https://github.com/adobe/react-spectrum/blob/09a258f098147f7e67a0430b593b08168220ec4a/packages/%40react-aria/select/src/useSelect.ts#L144
```
!props['aria-label'] && valueId,
```

If an aria-label is provided to the component, then we prefer that over the visible label, but we still fall back to the visible label in all other cases.
It also keeps the merging of the current value.

I think it matches the original description of the PR which introduced it as well https://github.com/adobe/react-spectrum/pull/306

**Comment by jguddas** [comment]
> If an aria-label is provided to the component, then we prefer that over the visible label, but we still fall back to the visible label in all other cases.

Definitly, there are 4 casse:

1. No aria props
Set `aria-labledby` to `value.id label.id`.
2. Only aria-label
Set `aria-labledby` to `value.id trigger.id` and trigger `aria-label` to `props['aria-label']`.
3. Only aria-labledby
Set `aria-labledby` to `value.id props['aria-labledby']`.
4. Both aria-label and aria-labledby
Set `aria-labledby` to `value.id props['aria-labledby']` and trigger `aria-label` to `props['aria-label']`.

So in the end the labledby is always value plus label.

**Comment by snowystinger** [comment]
Would you be willing to make a PR?

**Comment by jguddas** [comment]
> Would you be willing to make a PR?

I will take a look.

**Comment by LFDanLu** [comment]
@jguddas Upon testing this change futher today, the team decided to revert the change since it is actually a breaking change for many of our hooks/components that rely on the current aria-label + aria-labelledby chaining. There are some details here: https://github.com/adobe/react-spectrum/pull/5036#issue-1884735434, let us know what you think. For your own use case you could use https://github.com/adobe/react-spectrum/issues/4920#issuecomment-1681476500 or potentially ignore the majority of the `labelProps` returned by the hook (keeping onClick only)

**Comment by jguddas** [comment]
Should we reopen this or is it impossible to deploy a breaking changes at all and we just have to give up?

**Comment by LFDanLu** [comment]
Ideally we would have a solution that wouldn't break the RSP Picker or other people's hook implementation if they were relying on the prior `aria-label` + `aria-labelledby` chaining. The core issue here is that providing a `label` to the `useSelect` is a bit different from passing a label to a native select because we assume the `label` provided is going to be a visual label and thus we cannot omit the trigger's `aria-labelledby`. Since stuff from the hook can be modified freely by end users to suit their desired experience as mentioned in the workaround in some of the prior comments, the team was leaning towards keeping the hook as is. What do you think?
